core/receipt.go

```diff
@@ -62,22 +66,40 @@ func messagesSentHash(messages []*L2ToL1Message) *felt.Felt {
 
 func receiptCommitment(receipts []*TransactionReceipt) (*felt.Felt, error) {
 	var commitment *felt.Felt
-
 	return commitment, trie.RunOnTempTriePoseidon(commitmentTrieHeight, func(trie *trie.Trie) error {
-		for i, receipt := range receipts {
-			receiptTrieKey := new(felt.Felt).SetUint64(uint64(i))
-			_, err := trie.Put(receiptTrieKey, receipt.hash())
-			if err != nil {
-				return err
-			}
+		numWorkers := runtime.GOMAXPROCS(0)
+		receiptsPerWorker := max(1, len(receipts)/numWorkers)
+		workerPool := pool.New().WithErrors().WithMaxGoroutines(numWorkers)
+		var trieMutex sync.Mutex
+
+		for receiptIdx := 0; receiptIdx < len(receipts); receiptIdx += receiptsPerWorker {
+			startIdx := receiptIdx
+			endIdx := min(startIdx+receiptsPerWorker, len(receipts))
+			workerPool.Go(func() error {
+				for i, receipt := range receipts[startIdx:endIdx] {
+					receiptTrieKey := new(felt.Felt).SetUint64(uint64(receiptIdx + i))
+					receiptHash := receipt.hash()
+
+					trieMutex.Lock()
+					_, err := trie.Put(receiptTrieKey, receiptHash)
+					trieMutex.Unlock()
```

Comment:
I think current impl is not the most efficient one, we need to write benchmarks for that. You can add them in `receipt_pkg_test.go`. First for old impl, then for new one. After that I'll tell what else we can improve